[WIP]feat(social_media): Add feature to post on linkedin

gutsytechster opened the pull request.

gutsytechster committed:
feat(social_media): Add feature to post on linkedin

gutsytechster committed:
refactor(services.py): Create functions for making code modular

gutsytechster committed:
style(services.py): Add comments to upload_image_to_linkedin function

GeekyShacklebolt commented on nexus/celery.py:
```diff
@@ -37,4 +38,9 @@ def on_configure(self):
+app.conf.beat_schedule = {
+    'publish-posts': {
+        'task': 'publish_posts',
+        'schedule': timedelta(minutes=1),
```
We need to take this value of `minutes` from settings.

GeekyShacklebolt commented on nexus/social_media/tasks.py:
```diff
@@ -0,0 +1,9 @@
+from nexus.social_media.services import publish_post_service
+
+
+@app.task(name='publish_posts')
```
This can be `publish_posts_task` for better readability.

GeekyShacklebolt commented on nexus/social_media/services.py:
```diff
@@ -0,0 +1,166 @@
+# Third Party Stuff
+import requests
```
Did you run `pre-commit` hooks before making the commit? Should not `requests` be present in `Standard` imports?

gutsytechster replied:
Yeah I forgot to run `pre-commit`. Is there any workout that can be done now?

GeekyShacklebolt replied:
Fix it with another commit.

GeekyShacklebolt commented on nexus/social_media/services.py:
```diff
@@ -0,0 +1,166 @@
+from nexus.social_media.models import Post
+
+
+def update_post_object(post):
```
We don't need this function. We'll do all updates in `publish_posts_service`. We'll first update all those posts that are ready to get published and then will call functions to publish each post. If we update each post like this function is doing, we may break the working. Suppose there are 1000 posts needed to be published and you are publishing and updating each post, now you are done with 500 posts, and next celery beat get executed then the rest 500 posts that were not published and not updated will _again_ get in the async list.

gutsytechster replied:
But there could be this case:- We update all the posts that are ready to be published but when we actually publish the post, the API may break. That would lead to inconsistent data as there would be post which aren't published but data would show that they have been published.

GeekyShacklebolt replied:
In that case, we can update the `post` oppositely. Like removing its `is_published = true` to `false` and so on.

GeekyShacklebolt commented on nexus/social_media/services.py:
```diff
@@ -0,0 +1,166 @@
+
+def upload_image_to_linkedin(author, headers, post, api_url_base, linkedin):
+    """Upload image to linkedin for given post.
+    :param author: Owner of the post to be used for response POST data.
```
Leave a line above it.

GeekyShacklebolt commented on nexus/social_media/services.py:
```diff
@@ -0,0 +1,166 @@
+    :param post: Post model instance.
+    :param api_url_base: Base url for LinkedIn API.
+    :param linkedin: LINKEDIN_AUTH object taken from settings.
+    :returns: A valid HTTP response.
```
Leave a line after it.

GeekyShacklebolt commented on nexus/social_media/services.py:
```diff
@@ -0,0 +1,166 @@
+
+def post_to_linkedin(post_id):
+    """Function to post on linkedin
+    :param post_id: UUID of the post instance to be posted.
```
Please correct docstring syntax as said above.

GeekyShacklebolt commented on nexus/social_media/services.py:
```diff
@@ -0,0 +1,166 @@
+
+        asset = json_response_assets.get('value').get('asset')
+        with open(post.image.file.name, 'rb') as f:
+            response_upload_image = requests.post(uploadUrl, data=f.read(),
```
Just to confirm, isn't that `f.read()` will be equivalent to `post.image`?

gutsytechster replied:
No. `post.image` gives the path of the image w.r.t the `social_media` directory. An example of such path would be `social_media/post/9rCe0BL7S5CHlqPcuATY4g.png`

GeekyShacklebolt replied:
What about `post.image.file`?

gutsytechster replied:
`post.image.file` and `post.image.file.name` gives the full name of the file wrt to the home of your system.

storymode7 replied:
If you wish you can do:
```suggestion
            response_upload_image = requests.post(uploadUrl, data=post.image.file.read(),
```
and remove the with statement

GeekyShacklebolt commented on nexus/social_media/services.py:
```diff
@@ -0,0 +1,166 @@
+
+def appropriate_response_action(response, post):
+    """Function to perform appropriate action based on the response given.
+    :param response: A valid HTTP response to perform suitable action.
```
Here as well, as said above.

GeekyShacklebolt reviewed (changes requested):
Please find my reviews inline and add test cases. And make the PR to CuriousLearner/nexus. We may review it there itself. Also have a look at this [PR](https://github.com/CuriousLearner/nexus/pull/18)

gutsytechster committed:
refactor(social_media): Provide readability and refactor functions

This adds the update_post_object functionality within the
publish_post_service and provides the more readable task name.
It also refactors the docstrings attached with each function.

gutsytechster committed:
test(social_media): Add missing tests for services

gutsytechster committed:
refactor(social_media): Rename variable name and refactor settings

This changes variable name 'uploadUrl' to 'upload_url' according
to PEP guidelines and also add LINKEDIN_API_URL_BASE to settings

gutsytechster committed:
Merge branch 'master' of github.com:CuriousLearner/nexus into post_linkedin

gutsytechster committed:
fix(test_post_to_facebook.py): Change test name to remove redundancy

gutsytechster committed:
refactor(social_media): Provide more readable names and rename tests

This provides more readable and generic names to methods and rename
test files as well according to test methods.

gutsytechster committed:
refactor(social_media): Sort python imports and update docs

This updates the .env.sample and CHANGELOG.md files and also
sort python imports.

gutsytechster committed:
fix(.gitignore): Ignore all the files start with celerybeat-schedule

gutsytechster committed:
fix(requirements): Specify pytest version to 4.4.2

gutsytechster committed:
fix(*): Fix small bugs

* fix(services): Remove unnecessary if-else condition.
* fix(services, tests): Replace .apply_async() with .delay()
* refactor(settings): Shift LINKEDIN settings.

gutsytechster committed:
Merge branch 'master' into post_linkedin

gutsytechster committed:
refactor(social_media): Refactor the function to return only one value

gutsytechster committed:
Merge branch 'master' into 'post_linkedin'

gutsytechster committed:
Merge branch 'master' of github.com:CuriousLearner/nexus into post_linkedin

gutsytechster committed:
refactor(social_media/services): Update docstrings

gutsytechster committed:
refactor(social_media): Rename function and its references

gutsytechster committed:
fix(social_media/services.py): Skip 200 status code

storymode7 committed:
tests(): Improve test coverage and add coveralls (#28)

gutsytechster committed:
Merge branch 'master' into post_linkedin
